nomad/structs/actions.go

```diff
@@ -26,6 +26,19 @@ type ActionListResponse struct {
 	QueryMeta
 }
 
+type JobRunActionRequest struct {
+	JobID     string
+	TaskGroup string
+	Task      string
+	Action    string
+	AllocID   string
+	WriteRequest
+	QueryOptions
```

Comment:
`WriteRequest` is for writes, and `QueryOptions` is for reads. You should only set one of these.